src/compiler/emitter.ts

```diff
@@ -3902,11 +3902,25 @@ module ts {
                     }
                 });
             }
+            
+            function sortAMDModules(amdModules: {name: string; path: string}[]) {
+                // AMD modules with declared variable names goes first
+                return amdModules.sort((moduleA, moduleB) => {
+                    if (moduleA.name == moduleB.name) {
+                        return 0;
+                    } else if (moduleA.name == undefined) {

```

Comment:
nit: `!moduleA.name` instead of `moduleA.name == undefined` as an empty name is still an invalid name. an later on you do check for amdDependency.name.